WMFComponents/Sources/WMFComponents/Components/Year in Review/View Models/WMFYearInReviewViewModel.swift

```diff
@@ -701,15 +701,15 @@ public class WMFYearInReviewViewModel: ObservableObject {
                 slides.append(.standard(personalizedSlides.saveCountSlide ?? (primaryAppLanguage.isEnglishWikipedia ? englishReadingListSlide : collectiveSavedArticlesSlide)))
                 slides.append(.standard(personalizedSlides.editCountSlide ?? (primaryAppLanguage.isEnglishWikipedia ? englishEditsSlide : collectiveAmountEditsSlide)))
                 slides.append(.standard(personalizedSlides.viewCountSlide ?? (primaryAppLanguage.isEnglishWikipedia ? englishEditsBytesSlide : collectiveEditsPerMinuteSlide)))
-                slides.append(.contribution(personalizedSlides.donateCountSlideV3 ?? nonContributorSlide))
+                if shouldShowDonateButton { slides.append(.contribution(personalizedSlides.donateCountSlideV3 ?? nonContributorSlide)) }
```

Comment:
This may have been unclear, but I think if they edited in the last year, we do still want to show them the contributor slide (see [comment](https://phabricator.wikimedia.org/T402696#11221002)). So some sort of pseudocode / test steps like this might work:

```
if editCount > 0 {
   insert donateCountSlideV3
} else if donateCount > 0 and shouldShowDonateButton {
   insert donateCountSlideV3
} else if shouldShowDonateButton {
   // edit count == 0 and donate count == 0 if they made it this far
   insert nonContributorSlide
} else {
   no slide should be inserted
}
```